inf-terraform-aws/Jenkinsfile.template

```diff
@@ -0,0 +1,322 @@
+        if (statusTF != 0) {
+          error "Creation of tfvars failed!"
+        }
+        tfvarsjson = readJSON file: './terraform.tfvars.json'
+        tfconsulpath="stacks/v1/"+tfvarsjson['meta_computer_system_name']+"/"+tfvarsjson['name']+"/"+tfvarsjson['meta_environment_type']+"/terraform-state"
+        git_login_url  = "git@bitbucket.biscrum.com"
+        git_login_port = 7999
+        println tfconsulpath
+
+        def status = sh(script: '''
+          mkdir $HOME/.ssh && \
+          mv -f -v $GITLAB_COE $HOME/.ssh/id_rsa && \
+          chmod 700 ~/.ssh && \
+          chmod 400 ~/.ssh/id_rsa*''',
+          returnStatus: true) 
+        if (status != 0) {
+          error "Init failed!"
+        }
+      }
+    }
+  } else { print "Stage Init has been skipped."}
+}
+
+//will be removed
+def stageDestroy(def context) {
+  if (stackdestroy == true) {
+    stage('Destroy')
+    {
+      withCredentials([
+        string(credentialsId: awsAccessKeyId, variable: 'AWS_ACCESS_KEY_ID'),
+        string(credentialsId: awsSecretAccessKey, variable: 'AWS_SECRET_ACCESS_KEY'),
+        string(credentialsId: awsRegion, variable: 'AWS_DEFAULT_REGION'),
+        string(credentialsId: consulHttpAddr, variable: 'CONSUL_HTTP_ADDR'),
+        string(credentialsId: consulHttpToken, variable: 'CONSUL_HTTP_TOKEN')      
+       ])
+      {
+        withEnv(["CONSUL_PATH=${tfconsulpath}",
+          "NO_PROXY=127.0.0.1,10.0.0.0/8,10.183.175.157,localhost,boehringer.com,eu.boehringer.com,am.boehringer.com",
+          "TF_WORKSPACE=${tfvarsjson['meta_environment_type']}",
+          "GIT_LOGIN_URL=${git_login_url}",
+          "GIT_LOGIN_PORT=${git_login_port}"
+        ])
+        {
+          def status = sh(script: '''
+            eval \"\$(rbenv init -)\" && \
+            ssh -o StrictHostKeyChecking=no -T -p \$GIT_LOGIN_PORT \$GIT_LOGIN_URL || true  && \
+            make destroy''', 
+            returnStatus: true)
+          if (status != 0) {
+            error "Test failed!"
+          }
+        }
+      }
+    }
+  } else { print "Stage Destroy has been skipped."}
+}
+
+def stageReport(def context) {
+  if (stackdeploy == true) {
+    stage('Report')
+    {      
+      def status = sh(script: '''
+          make install-report ''', 
+          returnStatus: true)
+      if (status != 0) {
+        error "Report failed!"
+      }  
+      dir("reports/install") {
+        sh "ls"
+      }   
+      stash(name: "installation-test-reports-junit-xml-${context.componentId}-${context.buildNumber}", includes: 'reports/install/*.log, reports/install/data/inspec/post-install/*.xml' , allowEmpty: true)  
+    }
+  } else { print "Stage Report has been skipped."}
+}
+
+def stageSmoke(def context) {
+  if (stackdeploy == true) {
+    stage('Smoke')
+    {
+      withCredentials([
+        string(credentialsId: awsAccessKeyId, variable: 'AWS_ACCESS_KEY_ID'),
+        string(credentialsId: awsSecretAccessKey, variable: 'AWS_SECRET_ACCESS_KEY'),

```

Comment:
couldn't we use `kubernetes.io/basic-auth` to store both `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` in one secret.